go/libraries/doltcore/sqle/dtables/diff_iter.go

```diff
@@ -484,3 +484,66 @@ func maybeTime(t *time.Time) interface{} {
 	}
 	return nil
 }
+
+//------------------------------------
+// diffPartitionRowIter
+//------------------------------------
+
+var _ sql.RowIter = (*diffPartitionRowIter)(nil)
+
+type diffPartitionRowIter struct {
+	diffPartitions   *DiffPartitions
+	ddb              *doltdb.DoltDB
+	joiner           *rowconv.Joiner
+	currentPartition *sql.Partition
+	currentRowIter   *sql.RowIter
+}
+
+func NewDiffTableFunctionRowIterForSinglePartition(partition sql.Partition, ddb *doltdb.DoltDB, joiner *rowconv.Joiner) *diffPartitionRowIter {
```

Comment:
Don't forget to fix up this constructor function name, too. 

Since we're passing in a single partition and there's no longer another constructor function, I'd probably just name this one: `NewDiffPartitionRowIter`